components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImplTest.java

```diff
@@ -0,0 +1,336 @@
+
+        identityDatabaseUtil = mockStatic(IdentityDatabaseUtil.class);
+        SecretManagerImpl secretManager = mock(SecretManagerImpl.class);
+        SecretType secretType = mock(SecretType.class);
+        ActionMgtServiceComponentHolder.getInstance().setSecretManager(secretManager);
+        when(secretType.getId()).thenReturn("secretId");
+        when(secretManager.getSecretType(any())).thenReturn(secretType);
+        mockDBConnection();
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+        identityDatabaseUtil.close();
+    }
+
+    @AfterClass
+    public void wrapUp() throws Exception {
+
+        closeH2Database();
+    }
+
+    @Test(priority = 1)
+    public void testAddAction() throws ActionMgtException {
+
+        Action creatingAction = buildMockAction(
+                "PreIssueAccessToken",
+                "To configure PreIssueAccessToken",
+                "https://example.com",
+                buildMockBasicAuthentication("admin", "admin"));
+        action = serviceImpl.addAction(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getPathParam(), creatingAction,
+                tenantDomain);
+        Assert.assertNotNull(action.getId());
+        Assert.assertEquals(creatingAction.getName(), action.getName());
+        Assert.assertEquals(creatingAction.getDescription(), action.getDescription());
+        Assert.assertEquals(Action.Status.ACTIVE, action.getStatus());
+        Assert.assertEquals(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getActionType(),
+                action.getType().getActionType());
+        Assert.assertEquals(creatingAction.getEndpoint().getUri(), action.getEndpoint().getUri());
+        Assert.assertEquals(creatingAction.getEndpoint().getAuthentication().getType(),
+                action.getEndpoint().getAuthentication().getType());
+        Assert.assertEquals(creatingAction.getEndpoint().getAuthentication().getProperties().size(),
+                action.getEndpoint().getAuthentication().getProperties().size());
+    }
+
+    @Test(priority = 2, expectedExceptions = ActionMgtException.class,
+            expectedExceptionsMessageRegExp = "Unable to create an Action.")
+    public void testAddMaximumActionsPerType() throws ActionMgtException {
+
+        Action creatingAction = buildMockAction(
+                "PreIssueAccessToken",
+                "To configure PreIssueAccessToken",
+                "https://example.com",
+                buildMockBasicAuthentication("admin", "admin"));
+        action = serviceImpl.addAction(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getPathParam(), creatingAction,
+                tenantDomain);
+    }
+
+    @Test(priority = 3)
+    public void testGetActionsByActionType() throws ActionMgtException {
+
+        List<Action> actions = serviceImpl.getActionsByActionType(
+                Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getPathParam(), tenantDomain);
+        Assert.assertEquals(1, actions.size());
```

Comment:
Shall we assert the action attributes comes from `serviceImpl.getActionsByActionType()`